fhirzeug/generators/python_pydantic/static_files/tests/test_extension.py

```diff
@@ -81,3 +95,151 @@ def test_extension_in_a_domain_resource() -> None:
     data["extension"][0]["extension"][1]["extension"] = [{"url": "test/example", "valueInteger": 45}]  # type: ignore
     with pytest.raises(pydantic.ValidationError):
         r4.from_dict(data)
+
+
+def test_primitive_extension_object(
+    human_extension: r4.Extension, integer_extension: r4.Extension
+) -> None:
+    """Test PrimitiveExtension object."""
+    r4.PrimitiveExtension()
+    r4.PrimitiveExtension(id=ID)
+    r4.PrimitiveExtension(extension=[human_extension, integer_extension])
+    r4.PrimitiveExtension(id=ID, extension=[human_extension])
+
+    with pytest.raises(pydantic.ValidationError):
+        # Extension must be a list
+        r4.PrimitiveExtension(extension=human_extension)
+
+
+def test_primitive_extension_basic_usage(primitive_extension: r4.PrimitiveExtension):
+    """Test usage of a PrimitiveExtension."""
+    r4.Patient()
+    r4.Patient(gender="unknown")
+    r4.Patient(gender__extension=primitive_extension)
+    r4.Patient(gender="unknown", gender__extension=primitive_extension)
+
+
+def test_primitive_list_extension_usage(primitive_extension: r4.PrimitiveExtension,):
+    """Test usage of primitive list extension."""
+    # `given` field is Optional for HumanName
+    r4.HumanName()
+
+    # `given` is a `List` field
+    # -> So is `given` extension
+    r4.HumanName(given=[NAME])
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(given=NAME)
+
+    r4.HumanName(given__extension=[primitive_extension])
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(given__extension=primitive_extension)
+
+    # If `given` and `given__extension` are both provided, they must be of same length
+    r4.HumanName(given=[NAME], given__extension=[primitive_extension])
+    r4.HumanName(
+        given=[NAME, NAME], given__extension=[primitive_extension, primitive_extension],
+    )
+
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(
+            given=[NAME], given__extension=[primitive_extension, primitive_extension],
+        )
+
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(given=[NAME, NAME], given__extension=[primitive_extension])
+
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(
+            given=[None, NAME],
+            given__extension=[primitive_extension, None, primitive_extension],
+        )
+
+    # Not the same length because None values are not removed from list
+    with pytest.raises(pydantic.ValidationError):
+        r4.HumanName(given=[None, NAME, None], given__extension=[primitive_extension])
+
+    # Not the same length but a list is either empty, either filled with None values
```

Comment:
Not sure to understand the comment.
`r4.HumanName(given=[None, NAME, None], given__extension=[primitive_extension])` is not considered to be valid and that's why the test is expecting a `pydantic.ValidationError`

Reply:
I'm thinking about this case:

```python
    r4.HumanName(
        given=[], given__extension=[primitive_extension, primitive_extension]
    )
```

`given` is an empty array here, so it's explicit, and I wonder if we shouldn't raise an exception here because the length is different instead of converting it to `None`.

Reply:
That was also my first opinion for this case but it has been debated between Matthias, Julian and the FHIR Interpretation group and decided that we would accept this.

Side note : 
```py
r4.HumanName(given=[None], given__extension=[primitive_extension, primitive_extension])
```
is also considered as valid and `[None]` is stripped to `None`

Reply:
Ok. Thanks for the info.